docs/commands.rst

```diff
@@ -44,3 +45,177 @@ BUNDLE
+
+.. click:example::
+    import rastervision as rv
+
+    class ExampleExperiments(rv.ExperimentSet):
+       def get_experiment_builder(self):
+           # Experiment generation code here
+           pass
+
+       def exp_example(self):
+           experiment_builder = get_experiment_builder()
+
+           e = experiment_builder \
+               .with_root_uri(tmp_dir) \
+               .with_custom_config({
+                   'cogify': {
+                       'key': 'test',
+                       'config': {
+                           'uris': [(src_path, cog_path)],
+                           'block_size': 128
+                       }
+                   }
+               }) \
+               .build()
+
+           return e
+
+
+
+Configurating an Aux Command directly
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can configure the command configuration using the builder pattern directly. Aux Command builders all have the `with_root_uri` method, to set the root URI that will store command configuration, as well as the ``with_config`` method. This ``with_config`` method accetps **kwargs for configuration values.
+
+You can return one or more command configuration directly from an experiment method, as a single command configuration or a list of configs.
+
+Below is an example of an ExperimentSet that has one experiment method, that returns a configuration for a cogify command.
+
+.. click:example::
+    import rastervision as rv
+
+    class Preprocess(rv.ExperimentSet):
+       def exp_cogify(self):
+           root_uri = 's3://my-bucket/cogify'
+           uris = [('s3://my-bucket/original/some.tif', 's3://my-bucket/cogs/some-cog.tif')]
+
+           cmd_config = rv.CommandConfig.builder(rv.COGIFY) \
+                                        .with_root_uri(root_uri) \
+                                        .with_config(uris=uris,
+                                                     resample_method='bilinear',
+                                                     compression='jpeg') \
+                                        .build()
+
+           return cmd_config
+
+Running Aux Commands
+~~~~~~~~~~~~~~~~~~~~
+
+By default Aux Commands won't run without explicitly being run. That means
+
+.. code-block:: terminal
+
+   > rastervision -p example run local -e example.Preprocess
+
+Will not run the above Cogify command, however this will:
+
+.. code-block:: terminal
+
+   > rastervision -p example run local -e example.Preprocess cogify
+
+
+COGIFY
+^^^^^^
+
+The ``COGIFY`` command will turn GDAL-readable images and turn them into `Cloud Optimized GeoTiffs <https://www.cogeo.org/>`_.
+
+See the CogifyCommand entry in the :ref:`aux command api` API docs for configuration options.
+
+
+Custom Commands
+---------------
+
+Custom Commands allow advanced Raster Vision users to implement their own commands using the :ref:`plugins` architecture.
```

Comment:
The docs leave me confused about the relationship between aux commands and custom commands. I think an aux command is just a type of custom command that was created using a helper that lets you leave out some of the details of a custom command. This might make more sense if you introduced custom commands first with an example. Then the next section could explain aux commands (as a special case of custom commands) and show how the previous example could be written more simply as an aux command.

Reply:
Also, I'm confused about how aux commands are configured vs custom commands. Is it the same for both?